crates/rattler_conda_types/src/match_spec/condition.rs

```diff
@@ -0,0 +1,257 @@
+use std::fmt::Display;
+
+use nom::{
+    branch::alt,
+    bytes::complete::tag,
+    character::complete::{char, multispace0, multispace1},
+    sequence::{delimited, preceded},
+    IResult, Parser,
+};
+use serde::Serialize;
+
+use crate::match_spec::parse::matchspec_parser;
+
+#[derive(Debug, Clone, PartialEq, Serialize)]
+pub struct Statement {
+    pub prefix: String,
+    pub condition: Option<MatchSpecCondition>,
+}
+
+/// Represents a condition in a match spec, which can be a match spec itself or a logical combination
+#[derive(Debug, Clone, PartialEq, Serialize, Eq, Hash)]
+pub enum MatchSpecCondition {
```

Comment:
One slight issue I have with this approach is that some information is lost when converting from a string. E.g. converting back to a string loses the order and parenthesis.

Im wondering if we can structure this differently or add a wrapping struct that holds the original statement or something along those lines. Ideally we build a concrete syntax tree.

The reason this is important is that when we report the conditions in an unsat error you want the information to closely resemble the users input.

Reply:
We actually never parse the condition back to a string so this is not true.